### ISSUE TITLE
Accessibility: Fix Dropbox login screen

### ISSUE THREAD

**Issue by laurent22** [laurent22/joplin#11796]
> I have the app installed on almost all my devices & it works quite well, for the most part.
> The part which does not, which is what brings me here today, is authentication of my BrailleSense notetaker with Dropbox, to enable sync.
> All is well, until the code appears & I attempt to copy it.
> My BrailleSense sees this text field as a field, which cannot be copied from. Sve tried writing the code on another device but then I get the error that the code is either invalid or expired.

Ref: https://discourse.joplinapp.org/t/syncing-my-braillesense-notetaker-via-dropbox/43661

**Comment by personalizedrefrigerator** [comment]
For me, the most difficulty comes from the Dropbox website's "copy code" screen. The code is displayed in what TalkBack identifies as an `EditBox, disabled`.

I was able to copy the code from Dropbox by:
1. Moving TalkBack's focus to the edit box.
2. Opening the TalkBack menu with an L-shaped gesture.
3. Clicking "Copy last spoken phrase" in the menu.
4. Switching to Joplin.
5. Editing the edit field for the code in Joplin and pasting the last spoken phrase.
6. Backspacing the "EditBox, disabled" part of the phrase.

It looks like it might be possible for us to specify a custom redirect URL instead of the "copy code" screen:
> The redirect_uri is optional with the code flow - if unspecified, the authorization code is displayed on dropbox.com for the user to copy and paste to your app.

**Comment by personalizedrefrigerator** [comment]
I've reported the issue to Dropbox. This is the [Dropbox forum post](https://www.dropboxforum.com/discussions/101001017/oauth-copy-code-page-accessibility/821263).

**Comment by laurent22** [comment]
Thanks for checking! I was under the impression that this screen was created by us but indeed if it's on Dropbox that's a different issue. Let's wait a few days to see if they answer your post, otherwise we can indeed investigate the `redirect_uri` option.

**Comment by personalizedrefrigerator** [comment]
`expo` seems to [include a library](https://docs.expo.dev/guides/authentication/#dropbox) that handles OAuth with providers including Dropbox. Using this here might simplify Dropbox auth.
